lib/stacks/cluster-types.ts

```diff
@@ -41,7 +46,8 @@ export interface ClusterProvider {
  * ClusterAddOn is the interface to which all Cluster addons should conform.
  */
 export interface ClusterAddOn {
-    deploy(clusterInfo: ClusterInfo): void;
+    readonly name?: string;
+    deploy(clusterInfo: ClusterInfo): cdk.Construct | void;
```

Comment:
As I mentioned during standup, we will also need a Promise<Construct> here is a valid return type. However, it has a bigger impact, I will handle it myself.